src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,36 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<T> getAll(Class<? extends Machine> type) {
+        if (type.equals(Bulldozer.class)) {
+            return (List<T>) new BulldozerProducer().get();
+        } else if (type.equals(Excavator.class)) {
+            return (List<T>) new ExcavatorProducer().get();
+        } else if (type.equals(Truck.class)) {
+            return (List<T>) new TruckProducer().get();
+        }
+        return new ArrayList<>();
```

Comment:
Consider creating a local variable for the result of `new BulldozerProducer().get()`, `new ExcavatorProducer().get()`, and `new TruckProducer().get()` to avoid unchecked assignment warnings. Also, this will make your code cleaner and easier to maintain.